lib/Catalyst/Plugin/Session/State/MAC.pm

```diff
@@ -0,0 +1,115 @@
+package Catalyst::Plugin::Session::State::MAC;
+
+=head1 NAME
+
+Catalyst::Plugin::Session::State::MAC
+
+=cut
+
+=head1 DESCRIPTION
+
+Overrides the cookie session state in order to keep use the MAC as the real session ID while keeping the cookie session ID as well
+
+=cut
+
+use Moose;
+use namespace::autoclean;
+use pf::util;
+
+extends 'Catalyst::Plugin::Session::State::Cookie';
+
+our $VERSION = "0.01";
+
+=head2 get_session_id
+
+Get the session ID for the session
+In our case, if there is a MAC we use it
+Otherwise we take the value in the session cookie
+If all fails we generate a session ID
+
+=cut
+
+sub get_session_id {
+    my $c = shift;
+
+    my $mac = $c->portalSession->clientMac;
+    if(valid_mac($mac)){
+        $mac =~ s/\://g;
+        return $mac;
```

Comment:
Catalyst does not consider it valid.
A catalyst session ID can only contain hexadecimal characters